File: src/main/java/com/example/ex3_2_back/entity/Movie.java
package com.example.ex3_2_back.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.media.SchemaProperty;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Builder
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Table(name = "t_Movie")
public class Movie {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Schema(defaultValue = "862")
    Integer id;
    Boolean adult;
    Integer budget;
    String homepage;
    @Schema(defaultValue = "功夫熊猫")
    String originalLanguage;
    String originalTitle;
    Float popularity;
    String posterPath;
    String releaseDate;
    Integer revenue;
    Integer runtime;
    String status;
    String tagline;
    String title;
    Float voteAverage;
    Integer voteCount;
    String overview;
    Integer seenCount;
    @Schema(defaultValue = "不知道什么公司")
    String companyName;
    @Schema(defaultValue = "美国")
    String countryName;
}

File: src/main/kotlin/com/example/ex3_2_back/entity/Movie.kt
package com.example.ex3_2_back.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigInteger

@Entity
@Table(name = "movie")
class Movie (
    @Id
    @Column(name = "movie_id")
    var id: Int = 0,
    var adult: Boolean = false,
    var budget: Int = 0,
    var homepage: String = "",
    @Column(name = "original_language")
    var originalLanguage: String = "",
    @Column(name = "original_title")
    var originalTitle: String = "",
    var popularity: Float = 0.0f,
    @Column(name = "poster_path")
    var posterPath: String = "",
    @Column(name = "release_date")
    var releaseDate: String = "",
    var revenue: Int = 0,
    var runtime: Int = 0,
    var status: String = "",
    var tagline: String = "",
    var title: String = "",
    @Column(name = "vote_average")
    var voteAverage: Float = 0.0f,
    @Column(name = "vote_count")
    var voteCount: Int = 0,
    var overview: String = "",
)